app/main/controller/image_similarity_controller.py

```diff
@@ -11,30 +11,37 @@
 
 api = Namespace('image_similarity', description='image similarity operations')
 image_similarity_request = api.model('image_similarity_request', {
-  'url': fields.String(required=True, description='image URL to be stored or queried for similarity'),
+  'url': fields.String(required=False, description='image URL to be stored or queried for similarity'),
+  'doc_id': fields.String(required=False, description='image ID to constrain uniqueness'),
   'threshold': fields.Float(required=False, default=0.9, description='minimum score to consider, between 0.0 and 1.0 (defaults to 0.9)'),
   'context': JsonObject(required=False, default=[], description='context')
 })
 
 def _after_log(retry_state):
   app.logger.debug("Retrying image similarity...")
 
+def delete_record(params):
+  deleted = db.session.query(ImageModel).filter(ImageModel.doc_id==params['doc_id']).delete()
```

Comment:
Let's please be sure that `params['doc_id']` is not `None`.

Reply:
Otherwise we can delete a bunch of images :)

Reply:
Fair! Updated.